components/webui/client/src/pages/SearchPage/SearchState/useUpdateStateWithMetadata.ts

```diff
@@ -41,32 +47,45 @@ const useUpdateStateWithMetadata = () => {
             case PRESTO_SEARCH_SIGNAL.DONE:
                 updateSearchUiState(SEARCH_UI_STATE.DONE);
                 break;
-            case PRESTO_SEARCH_SIGNAL.FAILED:
+            case PRESTO_SEARCH_SIGNAL.FAILED: {
+                const errorMsg = resultsMetadata.errorMsg || "An error occurred during search";
+                const errorName = resultsMetadata.errorName || "Search Failed";
+
+                updateErrorMsg(errorMsg);
+                updateErrorName(errorName);
+
                 // Presto reports query cancellation as a failure, but we treat as a successful
                 // completion.
                 if (resultsMetadata.errorName === PRESTO_CANCEL_ERROR_NAME) {
                     updateSearchUiState(SEARCH_UI_STATE.DONE);
-
                     break;
                 }
                 updateSearchUiState(SEARCH_UI_STATE.FAILED);
-                notification.error({
-                    description: resultsMetadata.errorMsg || "An error occurred during search",
-                    duration: 15,
-                    key: `search-failed-${resultsMetadata._id}`,
-                    message: resultsMetadata.errorName || "Search Failed",
-                    pauseOnHover: true,
-                    placement: "bottomRight",
-                    showProgress: true,
-                });
+
+                // Error for guided UI is shown in drawer.
+                if (sqlInterface === PRESTO_SQL_INTERFACE.FREEFORM) {
+                    notification.error({
+                        description: errorMsg,
+                        duration: 15,
+                        key: `search-failed-${resultsMetadata._id}`,
+                        message: errorName,
+                        pauseOnHover: true,
+                        placement: "bottomRight",
+                        showProgress: true,
+                    });
+                }
```

Comment:
```suggestion
                if (sqlInterface === PRESTO_SQL_INTERFACE.FREEFORM) {
                    notification.error({
                        description: errorMsg,
                        duration: 15,
                        key: `search-failed-${resultsMetadata._id}`,
                        message: errorName,
                        pauseOnHover: true,
                        placement: "bottomRight",
                        showProgress: true,
                    });
                } else {
                    updateQueryDrawerOpen(true);
                }
```

How about showing the drawer automatically on error?